### PR TITLE
Swapmath tests

### PR DESCRIPTION
**Pull request by dianakocsis**
## Related Issue
Which issue does this pull request resolve?

## Description of changes

related to PROTO-140

### CODE REVIEW

File: test/libraries/SwapMath.t.sol [hunk 1]
```diff
@@ -183,8 +183,8 @@ contract SwapMathTest is Test, GasSnapshot {
     ) public pure {
         vm.assume(sqrtPriceRaw > 0);
         vm.assume(sqrtPriceTargetRaw > 0);
-        vm.assume(feePips > 0);
-        vm.assume(feePips < 1e6);
+        vm.assume(feePips >= 0);
+        vm.assume(feePips <= 1e6);
```
Comment (snreynolds) [inline test/libraries/SwapMath.t.sol:187]: doesnt work for exact out

Comment (snreynolds) [reply test/libraries/SwapMath.t.sol:187]: might need to expect something different for exact out and 100%

**Review by snreynolds — COMMENTED** [review]
lookin good

File: test/libraries/SwapMath.t.sol [hunk 2]
```diff
@@ -208,8 +213,14 @@ contract SwapMathTest is Test, GasSnapshot {
 
         // didn't reach price target, entire amount must be consumed
         if (sqrtQ != sqrtPriceTargetRaw) {
+            uint256 absAmtRemaining;
+            if (amountRemaining == type(int256).min) {
+                absAmtRemaining = uint256(type(int256).max) + 1;
+            } else if (amountRemaining < 0) {
+                absAmtRemaining = uint256(-amountRemaining);
+            }
```
Comment (hensha256) [inline test/libraries/SwapMath.t.sol:221]: does it not need an `else absAmtRemaining = amountRemaining` ?

Comment (dianakocsis) [reply test/libraries/SwapMath.t.sol:221]: yes! its because type(int256).min and type(int256).max do not have the same absolute value

Comment (dianakocsis) [reply test/libraries/SwapMath.t.sol:221]: so i had to convert it to uint256 first then add 1

Comment (dianakocsis) [reply test/libraries/SwapMath.t.sol:221]: but itll still do the else statement so its more clear!

**Review by hensha256 — APPROVED** [review]

**Pull request merged**